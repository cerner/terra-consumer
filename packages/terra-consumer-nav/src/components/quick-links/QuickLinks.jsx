import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './QuickLink.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The children to be displayed within the quicklink container.
   */
  children: PropTypes.node,
};

const defaultProps = {
  children: null,
};

const QuickLinks = ({
  children,
  ...customProps
}) => (
  <div {...customProps} className={cx('quicklinks-container', customProps.className)}>
    {children}
  </div>
);

QuickLinks.propTypes = propTypes;
QuickLinks.defaultProps = defaultProps;

export default QuickLinks;
