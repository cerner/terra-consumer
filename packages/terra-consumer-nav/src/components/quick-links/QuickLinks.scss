@import '../../variables';

:local {
  .quicklink {
    background: #fff;
    border: 0;
    border-radius: 3px;
    color: #3d3d3d;
    display: inline-block;
    font-family: inherit;
    font-size: 16px;
    line-height: 35px;
    margin-bottom: 10px;
    padding: 0;
    text-align: center;
    text-decoration: none;
    width: 100%;

    &:hover {
      background: #eee;
      cursor: pointer;
    }

    @media screen and (max-width: $terra-medium-breakpoint) {
      display: none;
    }
  }
}
