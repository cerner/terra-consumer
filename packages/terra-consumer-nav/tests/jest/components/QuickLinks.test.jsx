import React from 'react';
import QuickLinks from '../../../src/components/quick-links/QuickLinks';

describe('QuickLinks', () => {
  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(<QuickLinks />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a component with children', () => {
    const wrapper = shallow(
      <QuickLinks>
        <div>Item</div>
      </QuickLinks>,
      );
    expect(wrapper).toMatchSnapshot();
  });

  it('should apply custom classes', () => {
    const wrapper = shallow(<QuickLinks className="test-class" />);
    expect(wrapper).toMatchSnapshot();
  });
});
