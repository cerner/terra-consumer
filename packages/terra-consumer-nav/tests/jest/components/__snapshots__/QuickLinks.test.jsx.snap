// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickLinks should apply custom classes 1`] = `
<div
  className="quicklinks-container test-class"
/>
`;

exports[`QuickLinks should render a component with children 1`] = `
<div
  className="quicklinks-container"
>
  <div>
    Item
  </div>
</div>
`;

exports[`QuickLinks should render a default component 1`] = `
<div
  className="quicklinks-container"
/>
`;
